mm/kfence.c

```diff
@@ -79,6 +81,21 @@ static inline depot_stack_handle_t save_stack(gfp_t flags)
 	return stack_depot_save(entries, nr_entries, flags);
 }
 
+static inline void kfence_disable(void)
+{
+	pr_err("Disabling KFENCE\n");
+	WRITE_ONCE(kfence_enabled, false);
+}
+
+#define kfence_assert(cond)                                                    \
+	do {                                                                   \
+		if (unlikely(!(cond))) {                                       \
+			WARN(1, "KFENCE assertion failure at %s:%d\n",         \
+			     __FILE__, __LINE__);                              \
```

Comment:
WARN already has file and line number.

Why not just 'if (WARN(!(cond), "KFENCE assertion failure")' ?

Reply:
A simpler solution would be to just have an expression macro:

#define KFENCE_WARN_ON(cond) \({ \
   bool __cond = WARN_ON(cond); \
   if (__cond)\ 
     kfence_disable(); \
  __cond; \
})

Because every assertion failure or warning should go into an error-recovery path.

Reply:
Yes, you are right. I actually just blindly replaced BUG_ONs with the new assertion without thinking whether the recovery is needed. I'll fix that.